bigquery/client.py

```diff
@@ -253,6 +273,27 @@ def _submit_query_job(self, query_data):
 
         return job_id, [self._transform_row(row, schema) for row in rows]
 
+    def _get_job_reference(self, job_id):
+        """ Get job reference from job_id
+            For more details, see:
+            https://cloud.google.com/bigquery/docs/reference/rest/v2/jobs#resource
+
+        Parameters
+        ----------
+        job_id: 
+            Id of the job
+
+        Returns
+        -------
+        job_reference: json of job_reference       
+        """
+        job_reference = {
+            "projectId": self.project_id,
```

Comment:
Shouldn't this use the overridden `project_id` if it's passed into the function?

Reply:
Yeah, but it uses the self.project_id. What if the service account does not have project permission, which means it cannot authenticate to the project. There is a use case at my company like this:
- a service account is created, but all jobs is submitted to a default billing project_id. 
- This service account does not have project permission for other projects due to many sensitive datasets. In this case, the service account cannot authenticate to those projects. So the service account only has permission to access only a few datasets in that project if the datasets are shared.
- However, Google supports this use case which is the job can be submitted a billing project for billing purposes and tracking jobs, but it can process data in another dataset in another project which it is shared.

Hope it makes sense. I know it is a bit complicated, but I know there are a few companies have these restrictions to protect their data.

Best,
Tuan

Reply:
Yes, sorry, I understand the overall changes and it makes sense to me. I was specially referring to `_get_job_reference` and why it's hardcoded to use `self.project_id` instead of passing in a `project_id` and calling `_get_project_id(project_id)` like the rest of your changes?

Reply:
Oh, because the `_insert_job` always submit jobs to `self.project_id`, which is the billing project. Then `jobReference.projectId` will always be `self.project_id`. That is why `_get_job_reference` is hardcoded using the `self.project_id`. Hope it makes sense!
- References: https://cloud.google.com/bigquery/docs/reference/rest/v2/jobs